### ISSUE TITLE
azurerm_stream_analytics_output_eventhub: shared_access_policy_key and shared_access_policy_name required even though authentication_mode  = "Msi"

### ISSUE THREAD

**Issue by Jezevecek** [hashicorp/terraform-provider-azurerm#19442]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.3.2

### AzureRM Provider Version

3.33.0

### Affected Resource(s)/Data Source(s)

azurerm_stream_analytics_output_eventhub

### Terraform Configuration Files

```hcl
resource "azurerm_stream_analytics_output_eventhub" "example" {
  name                         = "input"
  stream_analytics_job_name    = "job"
  resource_group_name          = "rg-eh"
  eventhub_name                = "eh_name"
  servicebus_namespace         = "eh_namespace"
  authentication_mode          = "Msi"
  serialization {
    type      = "Json"
    encoding  = "UTF8" 
    format    = "LineSeparated"
  }
}
```

### Debug Output/Panic Output

```shell
Error: Missing required argument
..

The argument "shared_access_policy_key" is required, but no definition was found.

Error: Missing required argument
..

The argument "shared_access_policy_name" is required, but no definition was found.
```

### Expected Behaviour

No issues

### Actual Behaviour

Validating fails

### Steps to Reproduce

terraform validate

**Comment by jiaweitao001** [comment]
Fix PR https://github.com/hashicorp/terraform-provider-azurerm/pull/19447

**Comment by rcskosir** [comment]
@Jezevecek thank you for opening this issue and thank you @jiaweitao001 for tagging the fix.
This functionality has been released in [v3.34.0 of the Terraform Provider](https://github.com/hashicorp/terraform-provider-azurerm/blob/v3.34.0/CHANGELOG.md). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading. I will go ahead and close this issue.

For further feature requests or bug reports with this functionality, please create a [new GitHub issue](https://github.com/hashicorp/terraform-provider-azurerm/issues/new/choose) following the template. Thank you!
